File: 2078. Two Furthest Houses With Different Colors/solution.js
/**
 * @param {number[]} colors
 * @return {number}
 */
 var maxDistance = function(colors) {
  let furthestFromStart = colors.length - 1;
  let furthestFromEnd = 0;
  while (colors[furthestFromStart] == colors[0]){
    furthestFromStart -= 1;
  }
  while (colors[furthestFromEnd] == colors[colors.length - 1]){
    furthestFromEnd += 1;
  }

  return Math.max(colors.length - 1 - furthestFromEnd, furthestFromStart);
};

File: 2078. Two Furthest Houses With Different Colors/solution.cpp
class Solution {
public:
  int maxDistance(vector<int>& colors) {
    int furthestFromStart = colors.size() - 1;
    int furthestFromEnd = 0;
    while (colors[furthestFromStart] == colors[0]){
      furthestFromStart -= 1;
    }
    while (colors[furthestFromEnd] == colors[colors.size() - 1]){
      furthestFromEnd += 1;
    }       
    
    int max = colors.size() - 1 - furthestFromEnd;
    if (max < furthestFromStart){
      max = furthestFromStart;
    }
    
    return max;
  }
};